app/controllers/collections_controller.rb

```diff
@@ -127,6 +142,25 @@ def ensure_correct_collection_provider
     render :show
   end
 
+  def prepare_translated_collections
+    original_collection_native_xml = cmr_client.get_concept(params[:id],token, {})
+    return false if !original_collection_native_xml.success?
```

Comment:
In the case of these errors, instead of returning `false`, I think it might be better if you returned something like `{  error: 'this thing failed' }`. That way you can check for something more specific (an error key), you can provide a more specific error message since there are 3 places you do this, and it is a little more similar to how we deal with cmr responses (not necessarily the best reason).

Reply:
Great point. This makes the error handling much cleaner too